### ISSUE TITLE
test custom dataset on pretrained model

### ISSUE THREAD

**Issue by ranch-hands** [taesungp/contrastive-unpaired-translation#87]
hi
can you please provide pre trained models and explain thoroughly how exactly we can test the model on our customized dataset? 
I know you said we should create testA and testB folders, but where? and is there a way to just feed it with one set of images (not a couple set)?
I see there are so many networks(cut,fastcut,pix2pix, cyclegan,...) with many different datasets like edge to pic, night to day, zebra to horse,..but there is no thorough explanation for how precisely we should use them. 
PLEASE help and describe ,I'm so confused by now.

**Comment by George-Ogden** [comment]
Hi @ranch-hands 
If you want to test on a pretrained model, run
```
python test.py --dataroot DIRECTORY --checkpoints_dir CHECKPOINTS --name EXPERIMENT
```
So it should look something like this:
```
.
|-- DIRECTORY
|   |-- testA
|   |   |-- testA1.png
|   |   |-- testA2.jpg
|   |   ...
|   |-- testB
|   |   |-- testB1.png
|   |   |-- testB2.jpg
|   |   ...
|   |-- trainA
|   |   |-- trainA1.png
|   |   |-- trainA2.jpg
|   |   ...
|   `-- trainB
|   |   |-- trainB1.png
|   |   |-- trainB2.jpg
|   |  ...
|-- CHECKPOINTS
|   |-- EXPERIMENT
|   |   |-- latest_net_G.pth
|   |   |-- latest_net_F.pth

```
alongside all the other folders

**Comment by ranch-hands** [comment]
Hi @George-Ogden 
Thank you for your reply. Do you know what is the label for training and where is it?
I want to train the model on my own data set but I don't see any label for dataset.
Thanks again for your help

**Comment by ranch-hands** [comment]
Hi there

Thanks for your email. I really don't know why github doesn't show me the
posts that I asked a question or when a person answers me. by the way I
don't know how that network trains without a label. you may say that it is
an unsupervise network but I also don't understand when there is no label
how we compute loss of the network.

thanks alot
bests

**Comment by George-Ogden** [comment]
The labels for training are A and B, which allows you to choose your own classes. 
There are two types of losses - the Discriminator loss and the Cycle-Consistent loss, similar to a CycleGAN:

![image](https://user-images.githubusercontent.com/38294960/124079724-6bb9cc00-da41-11eb-94ae-4f57a5369314.png)

**Comment by ranch-hands** [comment]
Thank you for the clarification.
would you please introduce me some supervise network that changes the
picture or video to another one?


best regards

**Comment by George-Ogden** [comment]
Have a look at [pix2pix](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix) for paired image translation.

**Comment by ranch-hands** [comment]
HI @George-Ogden 
Do you know how these gan networks are evaluated and when we test them what do we evaluate?
for example in https://github.com/boschresearch/OASIS when we test the gan network, results will store in a folder with the same name but I just see a label and image and I dont know how I can say this result is good enough or not.
thanks inadvance

**Comment by George-Ogden** [comment]
Sorry, @ranch-hands I've never used this one before but I'll take a look.

**Comment by ranch-hands** [comment]
thanks dear, but I don't mean that particular repository. this is just
about gan networks that I really dont know how to evaluate their results.

**Comment by George-Ogden** [comment]
I highly recommend this lecture to understanding GANS, which looks at many of the things that it appears you have questions about: https://www.youtube.com/watch?v=wFsI2WqUfdA&ab_channel=DeepMind
